mlx_lm/tokenizer_utils.py

```diff
@@ -267,6 +267,30 @@ def __init__(
             if eos_token_ids is not None
             else {tokenizer.eos_token_id}
         )
+        self._think_start = None
+        self._think_end = None
+        self._tool_call_start = None
+        self._tool_call_end = None
+
+        THINK_TOKENS = [("<think>", "</think>")]
+        TOOL_CALL_TOKENS = [("<tool_call>", "</tool_call>")]
```

Comment:
Interesting! Do you have any thoughts on how we can make this better?

Reply:
Yes, for sure! I work on chat templates at Hugging Face and this is something we'd like as well; right now, the range of formats is very broad, and we do something basically like what you're doing here, we look for references to tools in the chat template to decide if the model supports tools or not. It's obviously hacky but there isn't really anything better!